src/main/java/EchoBot.java

```diff
@@ -0,0 +1,154 @@
+import java.awt.BorderLayout;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.util.ArrayList;
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.JTextArea;
+import javax.swing.JTextField;
+import javax.swing.SwingUtilities;
+import javax.swing.Timer;
+
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.FindCommand;
+import duke.command.MarkCommand;
+import duke.command.UnmarkCommand;
+import duke.storage.Storage;
+import duke.task.Task;
+import duke.ui.Ui;
```

Comment:
I think since you have changed Duke to echobot , you might consider renaming the package to EchoBot.

All in all, I think your codebase is very clean as I could easily track your code and never have to search up for the program flow. 👍